Sources/BraintreeCoreSwift/BTGraphQLHTTP.swift

```diff
@@ -0,0 +1,244 @@
+    }
+
+    // MARK: - Internal methods
+    
+    func httpRequest(
+        method: String,
+        parameters: NSDictionary? = [:],
+        completion: RequestCompletion?
+    ) {
+        var errorUserInfo: [String: Any] = [:]
+
+        if self.baseURL.absoluteString.isEmpty || self.baseURL.absoluteString == "" {
+            errorUserInfo["method"] = method
+            errorUserInfo["parameters"] = parameters
+            let error = Self.constructError(code: .missingBaseURL, userInfo: errorUserInfo)
+            completion?(nil, nil, error)
+            return
+        }
+        
+        let authorization: String
+        switch clientAuthorization {
+        case .authorizationFingerprint(let fingerprint):
+            authorization = fingerprint
+        case .tokenizationKey(let key):
+            authorization = key
+        default:
+            authorization = "" 
+        }
+        
+        guard let components = URLComponents(string: baseURL.absoluteString) else {
+            let error = Self.constructError(
+                code: .urlStringInvalid,
+                userInfo: [NSLocalizedDescriptionKey: "The URL absolute string is malformed or invalid."]
+            )
+            completion?(nil, nil, error)
+            return
+        }
+
+        guard let urlFromComponents = components.url else {
+            let error = Self.constructError(
+                code: .urlStringInvalid,
+                userInfo: [NSLocalizedDescriptionKey: "The URL absolute string is malformed or invalid."]
+            )
+            completion?(nil, nil, error)
+            return
+        }
+
+        let headers = [
+            "User-Agent": userAgentString(),
+            "Braintree-Version": BTCoreConstants.graphQLVersion,
+            "Authorization": "Bearer \(authorization)",
+            "Content-Type": "application/json; charset=utf-8"
+        ]
+        
+        var request: URLRequest
+    
+        do {
+            let bodyData = try JSONSerialization.data(withJSONObject: parameters ?? [:])
+            request = URLRequest(url: urlFromComponents)
+            request.httpBody = bodyData
+            request.allHTTPHeaderFields = headers
+            request.httpMethod = method
+            
+            let task: URLSessionTask = session.dataTask(with: request) { [weak self] data, response, error in
+                guard let self = self else { return }
+                self.handleRequestCompletion(data: data, response: response, error: error, completion: completion)
+            }
+
+            task.resume()
+        } catch {
+            completion?(nil, nil, error)
+        }
+    }
+
+    @objc(handleRequestCompletion:response:error:completionBlock:)
+    public func handleRequestCompletion(
+        data: Data?,
+        response: URLResponse?,
+        error: Error?,
+        completion: RequestCompletion?
+    ) {
+        guard let completion = completion else {
+            return
+        }
```

Comment:
```suggestion
        guard let completion = completion else { return }
```

If the guard's is just returning, we can make it a 1-liner. Thoughts?

Reply:
Agree, I'm a fan of the single line for these